src/traits.rs

```diff
@@ -57,3 +57,8 @@ pub trait Predict<R: Records, T> {
 pub trait PredictRef<R: Records, T> {
     fn predict_ref<'a>(&'a self, x: &'a R) -> T;
 }
+
+/// Predict with model into a mutable reference of targets.
+pub trait PredictInto<R: Records, T> {
+    fn predict_into<'a>(&'a self, x: &'a R, y: &mut T);
```

Comment:
`into` normally means that some value is consumed and consequently a new one produces, I think `PredictInplace` is more clear with function `predict_inplace`

Reply:
you should also return an error with null type like `-> Result<(), E>`. I would then `unwrap` the error in the `Predict` implementations for simpler use. We could then continue making the distinction more explicit with `predict` and `predict_unwrap`, but in most cases an error there is a logical error in the program.

Reply:
@bytesnake What type of errors do you envision being returned from prediction? Right now prediction doesn't return errors at all so I'm not sure whether we want to unwrap on possible prediction errors. For example if we throw parameter errors in `predict` I'm not sure about unwrapping that. I'm also not sure about returning Results from prediction traits, since current `predict` impls are infallible so we'd need to have a lot of useless unwraps.

Reply:
mainly for dimensionality errors, but note that the change is semantically. I would like to collect all variants of the error inside the corresponding error enum and avoid continuing sparkling asserts throughout the code. The behaviour is still the same

Reply:
Unwrapping dimensionality errors make sense (we could even have a common error type), but we need to ensure that all the errors are ones that should be unwrapped. Maybe we should enforce that predictions only return dimensionality errors?

Reply:
Can you guys create a separate issue for the dimensionality error handling. I am happy to do this one but someone else will have to do that.

Reply:
you could add a variant `MismatchNumLabels` with expected, observed number and return that for prediction. In the future we may also have categorical features where invalid values may return errors.

> Maybe we should enforce that predictions only return dimensionality errors?

yes we can add a separate enum for prediction errors

Reply:
> Can you guys create a separate issue for the dimensionality error handling. I am happy to do this one but someone else will have to do that.

yes sure perhaps just add a note to your issue?